libraries/Microsoft.Bot.Builder.Dialogs.Declarative/ComponentRegistration.cs

```diff
@@ -1,27 +1,74 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
+﻿using System;
 using System.Collections.Generic;
-using Microsoft.Bot.Builder.Dialogs.Debugging;
-using Microsoft.Bot.Builder.Dialogs.Declarative.Resolvers;
-using Newtonsoft.Json;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Bot.Expressions;
 
 namespace Microsoft.Bot.Builder.Dialogs.Declarative
 {
     public class ComponentRegistration
     {
-        public ComponentRegistration()
+        /// <summary>
+        /// Gets component registrations.
+        /// </summary>
+        /// <value>
+        /// Component registrations object which implement IComponentRegistration or derived interface.
+        /// </value>
+        public static Lazy<List<ComponentRegistration>> Registrations { get; private set; } = new Lazy<List<ComponentRegistration>>(() =>
```

Comment:
I think this method is worth a remarks section explaining what we are doing, something along the lines of 

> We recursively walk through the non-system dependencies in the application domain and load the assemblies. Once loaded, we inspect the types exposed by the assemblies in order to identify and register declarative components.